design-proposals/descheduler-support.md

```diff
@@ -0,0 +1,106 @@
+
+| Migration Strategy    | VMI is migrateable                    | VMI is not migrateable | Remarks                               |
+|-----------------------|---------------------------------------|------------------------|---------------------------------------|
+| None                  | N/A                                   | N/A                    | Always approve                        |
+| LiveMigrate           | Mark the VMI for eviction and approve | Deny                   |                                       |
+| LiveMigrateIfPossible | Mark the VMI for eviction and approve | Approve                |                                       |
+| External              | N/A                                   | N/A                    | Mark the VMI for eviction and approve |
+
+
+### Disruption Budget Controller
+`virt-launcher` pods have another layer of protection against evictions using [PDBs](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
+
+The controller is responsible for managing PDBs:
+
+The controller decides whether a VMI needs protection against eviction based on its migration strategy (defined on the VMI with a fallback to a cluster-wide definition):
+
+| Migration Strategy    | Requires a PDB                 | 
+|-----------------------|--------------------------------|
+| None                  | False                          |
+| LiveMigrate           | True                           |
+| LiveMigrateIfPossible | Only if the VMI is migrateable |
+| External              | True                           |
+
+### Evacuation Controller
+The controller looks for potential VMIs to evict and tries to migrate them to another node.
+
+### Migration Controller
+The controller manages the migration process initiated by the Evacuation controller.
+The controller migrates the `virt-launcher` pod from the source node to a target node.
+After the migration process is completed, the original `virt-launcher` pod is deleted from the source node and the eviction process ends.
+
+## Motivation
+Cluster administrators want their workloads to be balanced across the cluster.
+These workloads include KubeVirt VMs.
+
+## Goals
+1. Enable KubeVirt to work on existing K8s clusters that already have a descheduler deployed.
+2. Enable cluster admins to introduce a descheduler to an existing cluster with KubeVirt deployed.
+
+## Non Goals
+The descheduler should respect the two annotations included in this document.
+
+## Definition of Users
+- Cluster administrator
+
+## User Stories
+- As a cluster admin running a K8s cluster with a descheduler, I want to deploy and use KubeVirt.
+- As a cluster admin running a K8s cluster with KubeVirt, I want to deploy a descheduler to load balance all my workloads.
+- As a cluster admin I want to enable cluster-wide descheduler support.
+
+## Repos
+1. kubevirt/kubevirt
+2. kubevirt/hyperconverged-cluster-operator
```

Comment:
How is it relevant to this feature?

Reply:
An enabler (feature-gate) should be added to have support for the feature in deployments which use it to manage KubeVirt.